### PR TITLE
Feature/username-db-add Adds backend capability for providing usernames to users. 

### PR DESCRIPTION
**Pull request by speters0816**
Features:
- Adds email column to sql table (schema.sql) that can be used as login instead of user. This allows the user to change their username at will, but not their email.
  - The username cannot be longer than 32 characters long. Inserting a username was tested. The character limit works.
  - This feature breaks the current html login for registering the username. It should be changed to "email". 
- Added `auth.py` file containing authentication files for requiring login before viewing the index and loading the user information from the database into `g.user`. 
  - `g.user` can then be used to grab the username once the user is loads the index page after login. 
- Registered the auth.py file blueprint in the `__init__.py` file. This allows the code to be modular. Different packages dedicated to different parts of the web application.
- `auth.py` function `load_logged_in_user` loads the user information before any http request regardless of where the view function is called in the code.

### CODE REVIEW

File: flask-discord/flaskd/schema.sql [hunk 1]
```diff
@@ -4,7 +4,8 @@ DROP TABLE IF EXISTS siteViews;
 
 CREATE TABLE user (
 	id INTEGER PRIMARY KEY AUTOINCREMENT,
-	username TEXT UNIQUE NOT NULL,
+	email TEXT UNIQUE NOT NULL,
```
Comment (spyralspy01) [inline flask-discord/flaskd/schema.sql:7]: Great, we added functionality to have any username irrespective of having to have it unique

File: flask-discord/flaskd/auth.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import functools
+
+from flask import ( 
+        Blueprint, flash, g, redirect, render_template, request, session, url_for
+        )
+
```
Comment (spyralspy01) [inline flask-discord/flaskd/auth.py:6]: Just wanted to add. Good work!

**Review by spyralspy01 — APPROVED** [review]
everything works good work.

**Review by Sathvik-Chowdary-Veerapaneni — APPROVED** [review]
Good work Sam.

**Pull request merged**